File: KOMBAT/src/Game/Game.java
package Game;

import java.sql.Statement;
import java.util.ArrayList;
import java.util.List;

public class Game
{
	private enum GameState
	{
		start,
		buyHex,
		buyMinion,
		execute,
		gameOver,
	}

	private final StrategyExecutor executor;
	private final Merchant merchant;
	private final MinionStorage storage;
	private final List<Player> players;
	private final HexMap map;

	private GameState state;

	private int playerTurn;
	private int fullTurn;

	public Game(StartInfo info)
	{
		executor = new StrategyExecutor();
		merchant = new Merchant();
		storage = new MinionStorage();
		map = new HexMap(Config.MAP_WIDTH, Config.MAP_HEIGHT);

		//initialize local vars
		playerTurn = 0;
		fullTurn = 0;

		//initialize player
		players = new ArrayList<>();

		Player p1 = new Player(info.info1(), new Budget(), info.deck1());
		p1.initialize(storage, merchant, map);
		players.add(p1);

		Player p2 = new Player(info.info2(), new Budget(), info.deck2());
		p2.initialize(storage, merchant, map);
		players.add(p2);
	}

	public void start()
	{
		state = GameState.start;
	}

	public void update(PlayerIntent intent)
	{
		resolveIntent(intent);
		checkSwitchState();
	}

	private void checkSwitchState()
	{
		switch (state)
		{
			case start ->
			{

			}
		}
	}

	private void resolveIntent(PlayerIntent intent)
	{
		switch (state)
		{
			case start ->
			{
				if (intent.intent() == PlayerIntent.Intent.buyMinion)
				{
					if (players.get(playerTurn).spawnMinion(intent.hex(), intent.minion()))
						nextTurn();
				}
			}
		}
	}

	private void nextTurn()
	{
		if (++playerTurn > players.size())
		{
			playerTurn = 0;
			fullTurn++;
		}
	}


}

File: kombat-frontend/components/game/model/Game.ts
import {Config, PlayerInfo} from "../../../ttypes/type";
import {GameStateEnum} from "../../../ttypes/enums";
import {Player} from "../type/Player";
import {HexMap} from "../type/HexMap";
import {Minion} from "../type/Minion";

export class Game
{
    private cfg: Config

    readonly players: Player[];

    readonly map:HexMap;
    public minion: Minion[] = [];

    public gameState: GameStateEnum = GameStateEnum.empty;

    public team: number = 0;
    public turn: number = 0;

    constructor(cfg: Config, p1:PlayerInfo,p2:PlayerInfo,deck1:Minion[],deck2:Minion[]) {
        this.cfg = cfg;
        this.players = Array.of(
            new Player(p1,deck1),
            new Player(p2,deck2)
        );

        this.map = new HexMap(cfg.mapWidth,cfg.mapHeight);
    }


}